### ISSUE TITLE
Enhancement: graphviz_layout without drawing

### ISSUE THREAD

**Issue by jamestwebber** [Qiskit/rustworkx#917]
First off, I'm excited to have found this package and I'm hoping to try it out soon. Thanks for building it!

### What is the expected enhancement?

In `networkx` it is possible to get the _coordinates_ of a graphviz-provided layout without actually drawing the graph. This is nice because it provides access to the algorithms (e.g. `sfdp`) without constraining the user to their display methods. I typically use `nx.nx_agraph.graphviz_layout` to get coordinates for a graph and then plot the resulting graph with `matplotlib` where I have better control over the display options (if only because I'm more familiar with the API).

I found https://github.com/Qiskit/rustworkx/issues/280 which is somewhat related to this. In that issue the decision was to not re-implement the algorithms (makes sense) but leave the layout interface entirely within the drawing code.

It looks like the interface with graphviz is via python, so maybe this is as simple as ~copying~ emulating the code in [`networkx.nx_agraph`](https://networkx.org/documentation/stable/_modules/networkx/drawing/nx_agraph.html)?

**Comment by mtreinish** [comment]
Yeah, I like this idea it was something I was recently discussing in a Qiskit PR that was leveraging `graphviz_draw()` (see: https://github.com/Qiskit/qiskit-terra/pull/10208). Yeah we can use use similar code to `networkx.nx_agraph` with an appropriately documented optional dependency on pygraphviz. The other option would be to just call graphviz with subprocess and use the json output (or some other raw text format) and parse the layout information from that. I guess it really depends on whether we need the full pygraphviz representation or not.

**Comment by jamestwebber** [comment]
> The other option would be to just call graphviz with subprocess and use the json output (or some other raw text format) and parse the layout information from that.

I don't love this version because of the overhead of writing/reading the data. Unless graphviz can deal with streams?
